File: JAVASCRIPT/ComposicionAgregacion/Clase 30_09/JS/Banco.js
import Cuenta from "./Cuenta.js";
import Cliente from "./Cliente.js"

class Banco {

    #NIT;
    #razonSocial;
    #cuenta;

    constructor(NIT, razonSocial,cuenta) {
        this.#NIT = NIT;
        this.#razonSocial = razonSocial;
        this.#cuenta = cuenta;
    }

    getNIT(){
        return this.#NIT;
    }

    setNIT(NIT){
        this.#NIT = NIT;
    }


    getRazonSocial(){
        return this.#razonSocial;
    }

    setRazonSocial(razonSocial){
        this.#razonSocial = razonSocial;
    }

    getCuenta(){
        return this.#cuenta;
    }

    setCuenta(cuenta){
        this.#cuenta = cuenta;
    }
    
    esCliente(cliente) {
        return cliente instanceof Cliente; // Comprobación de instancia
    }


    consignarDinero(monto) {
        return this.#cuenta.consignarMsg(this.#cuenta.getNroCuenta(), monto);
    }
    

    toString(){
        return `Banco [NIT= ${this.#NIT}, razon Social = ${this.#razonSocial}, cuenta = ${this.#cuenta}]`
    }

    esCliente(cliente){
        let flag = false;
        if (cliente instanceof Cliente) {
            flag = true;
        }
        return flag; 
    }

}

export default Banco;

File: 20242/Clase 30_09/ComposicionAgregracion/Clase/Banco.java
public class Banco {


    //Atributos
    private String NIT;
    private String razonSocial;
    private Cuenta cuenta;


    //Constructor
    public Banco(String nIT, String razonSocial, Cuenta cuenta) {
        this.NIT = nIT;
        this.razonSocial = razonSocial;
        this.cuenta = cuenta;
    }

    //Getter y setter
    public String getNIT() {
        return NIT;
    }


    public void setNIT(String nIT) {
        this.NIT = nIT;
    }


    public String getRazonSocial() {
        return razonSocial;
    }


    public void setRazonSocial(String razonSocial) {
        this.razonSocial = razonSocial;
    }


    public Cuenta getCuenta() {
        return cuenta;
    }


    public void setCuenta(Cuenta cuenta) {
        this.cuenta = cuenta;
    }

    @Override
    public String toString() {
        return "Banco [NIT=" + NIT + ", razonSocial=" + razonSocial + ", cuenta=" + cuenta + "]";
    }

    public boolean esCliente(Cliente cliente){
        boolean flag = false;
        if (cliente instanceof Cliente) {
            flag = true;
        }
        return flag; 
    }
    
    


    
}